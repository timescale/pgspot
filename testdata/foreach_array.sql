DO LANGUAGE plpgsql $$
DECLARE
    desired_roles text[] := '{pg_checkpoint,pg_signal_backend,pg_read_all_stats,pg_stat_scan_tables}';
    desired_role text;
BEGIN
    FOREACH desired_role IN ARRAY desired_roles LOOP
        IF pg_catalog.to_regrole(desired_role) IS NOT NULL THEN
            EXECUTE format('GRANT %I TO tsdbadmin', desired_role);
        END IF;
    END LOOP;
END;
$$;
