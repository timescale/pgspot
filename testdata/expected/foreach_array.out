PS016: Unqualified function call: format at line 1

 Errors: 0 Warnings: 1 Unknown: 0 

